components/director/internal/domain/formation/repository.go

```diff
@@ -156,6 +165,89 @@ func (r *repository) ListByIDsGlobal(ctx context.Context, formationIDs []string)
 	return items, nil
 }
 
+// ListByIDs returns all Formations with id in `formationIDs` visible for the tenant from the context
+func (r *repository) ListByIDs(ctx context.Context, formationIDs []string) ([]*model.Formation, error) {
+	if len(formationIDs) == 0 {
+		return nil, nil
+	}
+
+	tnt, err := tenant.LoadFromContext(ctx)
+	if err != nil {
+		return nil, errors.Wrapf(err, "while loading tenant from context")
+	}
+
+	var entityCollection EntityCollection
+	err = r.lister.List(ctx, resource.Formations, tnt, &entityCollection, repo.NewInConditionForStringValues(idTableColumn, formationIDs))
+	if err != nil {
+		return nil, err
+	}
+
+	items := make([]*model.Formation, 0, entityCollection.Len())
+	for _, entity := range entityCollection {
+		formationModel := r.conv.FromEntity(entity)
+
+		items = append(items, formationModel)
+	}
+
+	return items, nil
+}
+
+// ListObjectIDsOfTypeForFormations returns all object IDs of type `objectType` for Formations with names in `formationNames`
+func (r *repository) ListObjectIDsOfTypeForFormations(ctx context.Context, tenantID string, formationNames []string, objectType model.FormationAssignmentType) ([]string, error) {
+	if len(formationNames) == 0 {
+		return nil, nil
+	}
+
+	persist, err := persistence.FromCtx(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	inCond := repo.NewInConditionForStringValues(nameColumn, formationNames)
+	args, _ := inCond.GetQueryArgs()
+
+	args = append(args, tenantID, objectType)
+	listObjectIDsOfTypeForFormationStatement := fmt.Sprintf(listObjectIDsOfTypeForFormation, inCond.GetQueryPart())
+	listObjectIDsOfTypeForFormationStatement = sqlx.Rebind(sqlx.DOLLAR, listObjectIDsOfTypeForFormationStatement)
```

Comment:
The SQL statement uses In condition and we do not know how many arguments will be passed. If we add the $ signs by ourselves at runtime it will be the same as using rebind.